### PR TITLE
Add a realm argument to "create an event"

### PR DESCRIPTION
**Pull request by domenic**
Needed for https://github.com/whatwg/html/issues/1959

### CODE REVIEW

**Comment by domenic** [comment]
I wonder if "fire an event" should pass along _target_'s relevant Realm. That would need some testing though (in theory testing every use of "fire an event").

**Comment by annevk** [comment]
IDL should really have some low-level operation to create classes that everything else is layered upon. This is just a mess.

I think _target_'s relevant Realm would always be correct. Would be annoying to require a _Realm_ argument everywhere.

File: dom.bs [hunk 1]
```diff
@@ -963,10 +964,15 @@ must be run, given the arguments <var>type</var> and <var>eventInitDict</var>:
...
-it, run these steps:</p>
+it, and optionally given a <a>Realm</a> <var>realm</var>, run these steps:</p>
 
 <ol>
- <li><p>Create a new object <var>event</var> using <var>eventInterface</var>.
+ <li>
+  <p>Create a new object <var>event</var> using <var>eventInterface</var>. If <var>realm</var>
+  is given, use that Realm; otherwise, use the default behavior defined in Web IDL.
+
+  <p class="note">As of the time of this writing Web IDL does not yet define any default behavior;
```
Comment (annevk) [inline dom.bs]: Should we make this class="XXX"?

**Comment by domenic** [comment]
> I think target's relevant Realm would always be correct. Would be annoying to require a Realm argument everywhere. 

I ran some tests and this is not interoperable -_-.

- Chrome uses target's relevant Realm
- Gecko might use target's relevant Realm, but since it does the currently-nonstandard-but-oft-debated thing where adopting an element changes its relevant Realm, it's not interoperable with Chrome
- Edge appears to not use the target's relevant Realm, but instead maybe the realm of the document in which target finds itself or something similar
- Haven't tested WebKit yet.

Let's leave that for a separate issue.

**Pull request merged**